ql/src/java/org/apache/hadoop/hive/ql/exec/vector/mapjoin/fast/VectorMapJoinFastHashTableLoader.java

```diff
@@ -294,7 +295,10 @@ public void load(MapJoinTableContainer[] mapJoinTables,
         LOG.info("Finished loading the queue for input: {} waiting {} minutes for TPool shutdown", inputName, 2);
         addQueueDoneSentinel();
         loadExecService.shutdown();
-        loadExecService.awaitTermination(2, TimeUnit.MINUTES);
+
+        if (!loadExecService.awaitTermination(2, TimeUnit.MINUTES)) {
+          throw new HiveException("Failed to complete the hash table loader. Loading timed out.");
```

Comment:
This will be a doubly wrapped HiveException

Reply:
@pvary Can you give pointers on how do I handle this?

Reply:
Catch and rethrow in the surrounding try-catch?,
```
 } catch(HiveException he) {
     throw he;
 }
```